e2-studio-star-rx72n-firmware/libs/rx_hcsr04/src/rx_hcsr04.c

```
@@ -1217,6 +1310,86 @@ static rx_err_t internal_measure_echo_pulse(rx_hcsr04_t* handle, uint32_t* durat
+ * @see rx_hcsr04_isr_get_duration() Reads ISR-captured timestamps
+ *
+ * @since Version 1.2.0 (Issue #296 - IRQ-based HC-SR04 measurement)
+ */
+static rx_err_t internal_measure_echo_pulse_irq(rx_hcsr04_t* handle, uint32_t* duration_us)
+{
+  if (handle == nullptr || duration_us == nullptr) {
+    return k_rx_err_null_ptr;
+  }
+
+  /* Arm the ISR state machine before trigger pulse */
+  const rx_err_t start_err = rx_hcsr04_isr_start((uint8_t)handle->echo_irq);
```

Clang (14.0.6)

[warning] variable 'start_err' is not initialized

(cppcoreguidelines-init-variables)